### ISSUE TITLE
⚠️ Data Leakage: Must not use test data when fitting MinMaxScaler()

### ISSUE THREAD

**Issue by shure-dev** [huseinzol05/Stock-Prediction-Models#126]
Probably, I found a serious error. 

If I'm correct, we cannot use any information from test data when preprocessing data.

**However, your code applied `fit_transform()` to train and test data.**

This means train data can contain information from test data and effects accuracy.

Please correct me if my idea is wrong, thank you.

**Comment by shure-dev** [comment]
This answer seems working well for this issue.

https://stackoverflow.com/questions/70923839/sklearn-preprocessing-with-a-rolling-window

**Comment by shure-dev** [comment]
Probably, also we have to care about stationarity, when we treat time series data
